tests/api/test_addon_uploads.py

```diff
@@ -1752,6 +1752,160 @@ def test_upload_localized_extension_json_overwrite(base_url, session_auth):
+    """Edit the version specific fields, i.e. 'release_notes', 'license, 'compatibility'"""
+    addon = payloads.edit_addon_details['slug']
+    request = requests.get(
+        url=f'{base_url}{_addon_create}{addon}',
+        headers={'Authorization': f'Session {session_auth}'},
+    )
+    # get the version id of the version we want to edit
+    version = request.json()['current_version']['id']
+    payload = payloads.edit_version_details
+    edit_version = requests.patch(
+        url=f'{base_url}{_addon_create}{addon}/versions/{version}/',
+        headers={
+            'Authorization': f'Session {session_auth}',
+            'Content-Type': 'application/json',
+        },
+        data=json.dumps(payload),
+    )
+    edit_version.raise_for_status()
+    response = edit_version.json()
+    # verify that the data we sent has been registered correctly in the response we get
+    assert payload['license'] == response['license']['slug']
+    assert payload['compatibility'] == response['compatibility']
+    assert payload['release_notes'] == response['release_notes']
+
+
+@pytest.mark.serial
+@pytest.mark.create_session('api_user')
+def test_edit_version_set_custom_license(base_url, session_auth):
+    """Instead of using a predefined addon license provided by AMO, add a
+    custom license with 'name' and 'text' defined by the addon author"""
+    addon = payloads.edit_addon_details['slug']
+    request = requests.get(
+        url=f'{base_url}{_addon_create}{addon}',
+        headers={'Authorization': f'Session {session_auth}'},
+    )
+    # get the version id of the version we want to edit
+    version = request.json()['current_version']['id']
+    payload = payloads.custom_license
+    edit_version = requests.patch(
+        url=f'{base_url}{_addon_create}{addon}/versions/{version}/',
+        headers={
+            'Authorization': f'Session {session_auth}',
+            'Content-Type': 'application/json',
+        },
+        data=json.dumps(payload),
+    )
+    edit_version.raise_for_status()
+    response = edit_version.json()
+    assert payload['custom_license']['name'] == response['license']['name']
+    assert payload['custom_license']['text'] == response['license']['text']
+
+
+@pytest.mark.serial
+@pytest.mark.create_session('api_user')
+def test_edit_version_invalid_license(base_url, session_auth):
+    """Extension license slugs have to match one of the predefined licenses accepted by AMO"""
+    addon = payloads.edit_addon_details['slug']
+    request = requests.get(
+        url=f'{base_url}{_addon_create}{addon}',
+        headers={'Authorization': f'Session {session_auth}'},
+    )
+    # get the version id of the version we want to edit
+    version = request.json()['current_version']['id']
+    # set a few unaccepted/invalid data types as license slugs
+    invalid_license = ['', None, False, 123, 'random slug', {'value': 'some value'}]
+    for slug in invalid_license:
```

Comment:
Any reason you didn't parametrize this?

Reply:
I think I didn't want to repeat the version id request for each case, but that shouldn't be too taxing on the api. I think I could switch to a parameterize form.